guides/git-workflow.md

```diff
@@ -1,21 +1,89 @@
+
+### The Branching Workflow <a name="branching-workflow"></a>
+
+1. Make a local copy of the `spacetelescope` repository by cloning the repository (e.g. `git clone https://github.com/username/repository_name.git`).  Note that, unless you explicitly delete your clone of the repository, this only has to be done once.
+
+2. Ensure that your clone is pointing to the `spacetelescope` remote repository with `git remote -v`.  You should see that `origin` corresponds to the URL from step (1).  If it doesn't, you can add it with `git remote add origin https://github.com/username/repository_name.git`.
+
+3. Create a branch on the clone to develop software changes on.  Branch names should be short but descriptive (e.g. `new-database-table` or `fix-ingest-algorithm`), and not too generic (e.g. `big-fix`).  Consistent use of hyphens is encouraged.
+    1. `git branch <branchname>`
+    2. `git checkout <branchname>` - you can use this command to switch back and forth between existing branches.
+    3. Perform local software changes using the nominal `git add`/`git commit -m` cycle:
+       1. `git status` -  allows you to see which files have changed.
+       2. `git add <new or changed files you want to commit>`
+       3. `git commit -m 'Explanation of changes you've done with these files'`
+
+4. Push the branch to the `spacetelescope` repository with `git push origin <branchname>`.
+
+5. In the `spacetelescope` repository, create a pull request for the recently pushed branch.  You will want to set the "base" option to point to the branch you would like to merge changes into (e.g. `master`), and the "compare" option to point to the new branch (i.e. `<branchname>`).  Note that if the branch is still under heavy development, you can put `WIP:` at the beginning of the pull request title to signify that the pull request is still a work in progress (i.e. `WIP: Example Pull Request Title`).  Not until the `WIP:` tag is explicitly removed will the pull request be deemed 'mergeable'.
+
+6. Assign the pull request a reviewer, selecting a maintainer of the `spacetelescope` repository.  They will review your pull request and either accept the request and merge, or ask for additional changes.
+
+7. Iterate with your reviewer(s) on additional changes if necessary, addressing any comments on your pull request.  If changes are required, you may end up iterating over steps 3.ii, 3.iii and 4 several times while working with your reviewer.
+
+8. Once the pull request has been accepted and merged, you can delete your local branch with `git branch -d <branchname>`.
```

Comment:
Is deleting the branch recommended for this workflow? I know people from both sides and they usually feel strongly one way or another. What is STScI's stance on this? Do we have one?

Reply:
I would recommend it because there's no use for the branch anymore, but I don't think there's an official verdict on this.

Reply:
Yeah, i'm not sure that there is an official recommendation. I suggest we purposely leave it vague for now.